src/server/page_server.hpp

```diff
@@ -59,19 +61,34 @@ class page_server
     void finalize_page_fetcher ();
 
   private:
-    using active_tran_server_conn_t =
-	    cubcomm::request_sync_client_server<page_to_tran_request, tran_to_page_request, std::string>;
-
-    void receive_boot_info_request (cubpacking::unpacker &upk);
-    void receive_log_prior_list (cubpacking::unpacker &upk);
-    void receive_log_page_fetch (cubpacking::unpacker &upk);
-    void receive_data_page_fetch (cubpacking::unpacker &upk);
-    void receive_disconnect_request (cubpacking::unpacker &upk);
-
-    void on_log_page_read_result (const LOG_PAGE *log_page, int error_code);
-    void on_data_page_read_result (const FILEIO_PAGE *page_ptr, int error_code);
 
-    std::unique_ptr<active_tran_server_conn_t> m_active_tran_server_conn;
+    class connection_handler
+    {
+	using tran_server_conn_t =
+		cubcomm::request_sync_client_server<page_to_tran_request, tran_to_page_request, std::string>;
+
+      public:
+	connection_handler () = default;
```

Comment:
By replacing m_ps with a reference, there can be no default constructor.
```suggestion
	connection_handler () = delete;
```